### PR TITLE
test(router): Add tests for initialNavigation: enabledBlocking

### LINKED ISSUE #44355: Router initializer never completes if initial navigation fails
### Which @angular/* package(s) are relevant/releated to the feature request?

router

### Description

Rendering an app in SSR (using nguniversal) with an invalid URL (e.g. http://localhost:4200/electronics-spa/en/USD/Brands/Canon/c/brand_10%20or%20(1,2)=(select*from(select%20name_const(CHAR(82,88,106,99,113,78,74,70,73,118,87),1),name_const(CHAR(82,88,106,99,113,78,74,70,73,118,87),1))a)%20--%20and%201%3D1) causes the render to hang, and never release the taken resources.

The reason is that the navigation never completes on error, when the `Router` option `initialNavigation` is set to `enabled`.

### Proposed solution

Expose `RouterInitializer`.
EDIT: or make the router complete the navigation on error, if that makes sense.

### Alternatives considered

We came up with this _workaround_, but it would be nice if `ɵangular_packages_router_router_h` could be exposed.

```ts
import { Injectable, Injector, Provider } from '@angular/core';
import {
  NavigationError,
  Router,
  ROUTER_CONFIGURATION,
  ɵangular_packages_router_router_h as RouterInitializer,
} from '@angular/router';
import { filter, take } from 'rxjs/operators';

@Injectable()
export class CustomRouterInitializer extends RouterInitializer {
  appInitializer(): Promise<any> {
    const injector: Injector = (this as any).injector;
    const opts = injector.get(ROUTER_CONFIGURATION);

    if (
      opts.initialNavigation ===
      'enabled' /* for Ng11: || opts.initialNavigation === 'enabledBlocking' */
    ) {
      let resolve: any;
      const res = new Promise((r) => (resolve = r));

      injector
        .get(Router)
        .events.pipe(
          // We want to explicitly react on NavigationError event occurs just
          // after the first navigation start event
          take(2),
          filter((event) => event instanceof NavigationError)
        )
        .subscribe(() => resolve(false));

      super.appInitializer().then((val) => resolve(val));
      return res;
    }

    return super.appInitializer();
  }
}

export const routerFixProvider: Provider = {
  provide: RouterInitializer,
  useClass: CustomRouterInitializer,
};
```

The above workaround is for ng12. 
I noticed that in ng13 there's no similar private API available, unless I'm missing something.

### PR DESCRIPTION
**Pull request by atscott**
Adds additional tests to verify `enabledBlocking` functionality. The
initial attempt to fix #44355 would have broken the scenario where a new
navigation cancels the initial navigation. We also cannot rely on
`NavigationError` like the example workaround in that issue report is
doing because there won't be one if a guard simply rejects
the initial nav (#16211).

### CODE REVIEW

File: packages/router/test/bootstrap.spec.ts [hunk 1]
```diff
@@ -76,6 +77,119 @@ describe('bootstrap', () => {
     }
   }));
 
+  // TODO(#44355)
```
Comment (AndrewKushnir) [inline packages/router/test/bootstrap.spec.ts:80]: nit: would it make sense to provide a bit of info here as well for the context (on what the problem is)?

File: packages/router/test/bootstrap.spec.ts [hunk 2]
```diff
@@ -76,6 +77,119 @@ describe('bootstrap', () => {
...
+            router.events.subscribe(e => log.push(e.constructor.name));
+          }
+        }
+
+        await platformBrowserDynamic([]).bootstrapModule(TestModule).then(res => {
+          const router = res.injector.get(Router);
+          expect(router.navigated).toEqual(false);
+          expect(router.getCurrentNavigation()).toBeNull();
+          expect(log).toContain('NavigationError');
+          expect(log).toContain('TestModule');
```
Comment (AndrewKushnir) [inline packages/router/test/bootstrap.spec.ts:112]: super nit: I was confused why the "TestModule" comes after the "NavigationError", but we don't check the order, but only a presence of an entry in the log. May be swap those two lines?

File: packages/router/test/bootstrap.spec.ts [hunk 3]
```diff
@@ -76,6 +77,119 @@ describe('bootstrap', () => {
...
+      ],
+      declarations: [RootCmp],
+      bootstrap: [RootCmp],
+      providers: [...testProviders],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA]
+    })
+    class TestModule {
+      constructor(router: Router) {
+        log.push('TestModule');
+        router.events.subscribe(e => log.push(e.constructor.name));
```
Comment (AndrewKushnir) [inline packages/router/test/bootstrap.spec.ts:143]: I don't think we do any asserts based on the events in the log. Can we remove this line?

File: packages/router/test/bootstrap.spec.ts [hunk 4]
```diff
@@ -76,6 +77,119 @@ describe('bootstrap', () => {
...
```
Comment (AndrewKushnir) [inline packages/router/test/bootstrap.spec.ts:181]: This line is likely unused/unnecessary.

**Review by AndrewKushnir — APPROVED** [review]
👍

**Comment by jessicajaniuk** [comment]
There are merge issues with `13.3.x`. Switching to `target: rc`.

**Pull request closed without merging**